webapi/src/main/scala/org/knora/webapi/responders/admin/ListsResponder.scala

```diff
@@ -178,63 +177,37 @@ final case class ListsResponder(
    *
    * @param nodeIri        the Iri if the required node.
    * @return a [[ListItemGetResponseADM]].
    */
-  private def listGetRequestADM(nodeIri: IRI) = {
+  def listGetRequestADM(nodeIri: IRI): Task[ListItemGetResponseADM] = {
 
     def getNodeADM(childNode: ListChildNodeADM): Task[ListNodeGetResponseADM] =
       for {
         maybeNodeInfo <- listNodeInfoGetADM(nodeIri = nodeIri)
+        nodeInfo <- maybeNodeInfo match {
+                      case Some(childNodeInfo: ListChildNodeInfoADM) => ZIO.succeed(childNodeInfo)
+                      case _                                         => ZIO.fail(NotFoundException(s"Information not found for node '$nodeIri'"))
+                    }
+      } yield ListNodeGetResponseADM(NodeADM(nodeInfo, childNode.children))
 
-        nodeinfo = maybeNodeInfo match {
-                     case Some(childNodeInfo: ListChildNodeInfoADM) => childNodeInfo
-                     case _                                         => throw NotFoundException(s"Information not found for node '$nodeIri'")
-                   }
-
-        // make a NodeADM instance
-        entirenode = ListNodeGetResponseADM(
-                       node = NodeADM(
-                         nodeinfo = nodeinfo,
-                         children = childNode.children
-                       )
-                     )
-      } yield entirenode
-
-    for {
-      exists <- rootNodeByIriExists(nodeIri)
-      // Is root node IRI given?
-      result <-
-        if (exists) {
-          for {
-            // Yes. Get the entire list
-            maybeList <- listGetADM(rootNodeIri = nodeIri)
-
-            entireList = maybeList match {
-                           case Some(list) => ListGetResponseADM(list = list)
-                           case None       => throw NotFoundException(s"List '$nodeIri' not found")
-                         }
-          } yield entireList
-        } else {
-          for {
-            // No. Get the node and all its sublist children.
-            // First, get node itself and all children.
-            maybeNode <- listNodeGetADM(nodeIri = nodeIri, shallow = true)
-
-            entireNode <- maybeNode match {
-                            // make sure that it is a child node
-                            case Some(childNode: ListChildNodeADM) =>
-                              // get the info of the child node
-                              getNodeADM(childNode)
-
-                            case _ => throw NotFoundException(s"Node '$nodeIri' not found")
-                          }
-          } yield entireNode
-        }
-    } yield result
+    ZIO.ifZIO(rootNodeByIriExists(nodeIri))(
+      listGetADM(nodeIri).someOrFail(NotFoundException(s"List '$nodeIri' not found")).map(ListGetResponseADM.apply),
+      for {
+        // No. Get the node and all its sublist children.
+        // First, get node itself and all children.
```

Comment:
this comment "responds" to an earlier comment that no longer exists. I'd suggest deleting it.

Reply:
```suggestion
```